Possibility to add a new administration page

PetrHeinz opened the pull request:
Implemented using KnpMenuBundle, with a example commit marked with "TO BE DROPPED" which should NOT be merged, it's only there to demonstrate the added functionality.

| Q             | A
| ------------- | ---
|Description, reason for the PR| There was no way to add a new admin page easily
|New feature| Yes <!-- Do not forget to update CHANGELOG.md and possibly docs/ -->
|BC breaks| Yes <!-- Do not forget to update UPGRADE.md -->
|Fixes issues| ...
|Standards and tests pass| Yes
|Have you read and signed our [License Agreement for contributions](https://www.shopsys.com/license-agreement)?| Yes

TomasLudvik commented on packages/framework/src/Resources/config/services.yml:
```diff
@@ -91,19 +92,21 @@ services:
+  Shopsys\FrameworkBundle\Model\AdminNavigation\BreadcrumbOverrider: ~
+
+  Shopsys\FrameworkBundle\Model\AdminNavigation\SideMenuBuilder:
+    public: true
```
why is this public?

PetrHeinz replied:
The menu service must be public as it will be retrieved at runtime to keep it lazy-loaded.

https://symfony.com/doc/master/bundles/KnpMenuBundle/menu_builder_service.html

TomasLudvik commented on docs/index.md:
```diff
@@ -13,6 +13,7 @@
 * [Components](./introduction/components.md)
 * [Abstract Filesystem](./introduction/abstract-filesystem.md)
 * [Product Feeds](./introduction/product-feeds.md)
+* [Administration Menu](./introduction/administration-menu.md)
```
This should not be here ( 323555c8315afb8be006d3bbc46f4229d65dcda3 )

TomasLudvik commented on packages/framework/src/Resources/scripts/admin/sideMenu.js:
```diff
@@ -0,0 +1,24 @@
+(function ($) {
```
This commit ( 361fc88a84045b47a74978f243bb2f2a95694742 ) should be before the one with documentation

PetrHeinz replied:
It is (and was). GitHub displays the commits in different order, try fetching it and running `git ll` locally

PetrHeinz commented on packages/framework/src/Model/AdminNavigation/BreadcrumbOverrider.php:
```diff
@@ -0,0 +1,35 @@
+    /**
+     * @param string $lastItemLabel
+     */
+    public function overrideLastItem(string $lastItemLabel): void
```
I did consider it, but in the end the fact that this name is shorter and that this could be extended by more optional parameters without breaking BC won. I think it's still understandable very easily.

PetrHeinz commented on packages/framework/src/Model/AdminNavigation/RoutingExtension.php:
```diff
@@ -0,0 +1,54 @@
+ */
+class RoutingExtension implements ExtensionInterface
+{
+    protected $generator;
```
Fixed in 701329ed1

PetrHeinz commented:
Thanks for the review @TomasLudvik,  I've addressed all issues you've mentioned, can you take a look at it? I've rebased the branch (moving thing between commits and sorting the dependencies) and added a fixup commit for the code that actually changed.
